src/test/java/com/amazonaws/services/sqs/AmazonSQSVirtualQueuesClientIT.java

```diff
@@ -108,4 +107,25 @@ public void virtualQueueShouldNotExpireDuringLongReceive() throws InterruptedExc
         // Delete the queue so we don't get a spurious message about it expiring during the test shutdown
         client.deleteQueue(virtualQueueUrl);
     }
+
+    @Test
+    public void missingMessageAttributeIsReceivedAndDeleted() throws InterruptedException {
+        CreateQueueRequest request = new CreateQueueRequest()
+                .withQueueName("ShortLived")
+                .addAttributesEntry(AmazonSQSVirtualQueuesClient.VIRTUAL_QUEUE_HOST_QUEUE_ATTRIBUTE, hostQueueUrl)
+                .addAttributesEntry(AmazonSQSIdleQueueDeletingClient.IDLE_QUEUE_RETENTION_PERIOD, "10");
+        String virtualQueueUrl = client.createQueue(request).getQueueUrl();
+
+        ReceiveMessageRequest receiveRequest = new ReceiveMessageRequest()
+                .withQueueUrl(virtualQueueUrl)
+                .withWaitTimeSeconds(5);
+        SendMessageRequest sendMessageRequest = new SendMessageRequest()
+                .withQueueUrl(hostQueueUrl)
+                .withMessageBody("Missing Message attributes!")
+                .withDelaySeconds(5);
+
+        client.sendMessage(sendMessageRequest);
+        // Message sent with missing attribute is deleted
+        assertEquals(0, client.receiveMessage(receiveRequest).getMessages().size());
```

Comment:
Instead of checking if messages are received, can we check the number of messages available with a queue attributes call?

Reply:
Well, when sending a message without the virtual queue attribute will still work since it'll default to using the normal sqs sdk. The issue is when receiving the message that has missing attributes, so when calling receive, we make sure here that the message got deleted since receive returns no messages